### ISSUE TITLE
"Warmup" period for RNN training?

### ISSUE THREAD

**Issue by davebs** [fchollet/keras#1042]
Is there any concept of a warmup period for RNN training in Keras?

For instance, say you have a 100 timestep sample and you're making predictions at each timestep. Because each prediction _should_ act on long term dependencies, and there are no long term dependencies at the first timestep, the output for the first ~20 timesteps may be assumed to be garbage. 

In that case, the warmup period would be 20. Is there a way to remove those first 20 timesteps from the loss computations? 

Prediction is easier as you just slice the piece you want to keep, but I think training could benefit from this?

**Comment by davebs** [comment]
Moving in the direction of a custom cost function, I was thinking all I needed was:

    y_pred = y_pred[:, warmup:]
    y_true = y_true[:, warmup:]

at the top of the normal CCE cost function. This isn't working though (it starts to train but outputs -0.000 loss on train and 0.000 on validation. Am I misunderstanding the shape of y_pred and y_true?

**Comment by EderSantana** [comment]
you could use `sample_weight` and set to zero the values of the warmup you don't want to contribute to the cost function.
